src/measure_theory/bochner_integration.lean

```diff
@@ -1323,11 +1323,11 @@ lemma integral_mono_ae {f g : α → ℝ} (hf : integrable f μ) (hg : integrabl
   ∫ a, f a ∂μ ≤ ∫ a, g a ∂μ :=
 le_of_sub_nonneg $ integral_sub hg hf ▸ integral_nonneg_of_ae $ h.mono (λ a, sub_nonneg_of_le)
 
-lemma integral_mono {f g : α → ℝ} (hf : integrable f μ) (hg : integrable g μ) (h : f ≤ g) :
+@[mono] lemma integral_mono {f g : α → ℝ} (hf : integrable f μ) (hg : integrable g μ) (h : f ≤ g) :
   ∫ a, f a ∂μ ≤ ∫ a, g a ∂μ :=
 integral_mono_ae hf hg $ eventually_of_forall h
 
-lemma integral_mono_of_nonneg {f g : α → ℝ} (hf : 0 ≤ᵐ[μ] f) (hgi : integrable g μ)
+lemma integral_mono_ae_of_nonneg {f g : α → ℝ} (hf : 0 ≤ᵐ[μ] f) (hgi : integrable g μ)
   (h : f ≤ᵐ[μ] g) : ∫ a, f a ∂μ ≤ ∫ a, g a ∂μ :=
```

Comment:
The point of this lemma is to provide a statement that works when `f` is not necessarily integrable.  At the moment it is used only once in the library, in the lemma right after (`norm_integral_le_integral_norm`).  About half of your PR is filling in the variants of this lemma in all the variant situations, i.e. lots of `*_of_nonneg` everywhere.

Are you sure this is an important enough lemma that we want all these variants?  I can think of other versions that (well, if they're true ....) might be just as likely to help in situations when we don't have a priori integrability for `f`.  For example, perhaps it also works to assume `(hg' : 0 ≤ᵐ[μ] g)` instead of `hf`?

Maybe it's better just to let the user deal with it ad hoc when they don't have an integrability hypothesis for `f`.

Reply:
Yes I'm okay with that

Reply:
To try to express this more briefly:  I think this is kind of an ugly lemma, I think it's ok to have it once in a specific situation when it's used right after, but I'm not convinced it's generally useful, and I'm sad to see it proliferating :)

Reply:
> Yes I'm okay with that

OK, thanks!  And in case it wasn't clear, this is only about the `*_of_nonneg` lemmas.  The `*_nonneg` lemmas, where we have a _conclusion_ `0 ≤ ∫ ...`, are good and useful.